app/services/visit_request/cancel.rb

```diff
@@ -6,10 +6,23 @@ def initialize(visit_request)
 
     def call
       visit_request.canceled!
+      attend_first_user_from_waitlist
     end
 
     private
 
     attr_reader :visit_request
+
+    def attend_first_user_from_waitlist
+      if waiting_list_request = VisitRequest.where(waiting_list: true).first
```

Comment:
1. First `if` can be replaced with [guard clause](https://devblast.com/b/what-are-guard-clauses)
2. `VisitRequest.where(waiting_list: true).first` -> `VisitRequest. waiting_list.first`
3. `VisitRequest. waiting_list.first` -> we should supply the ordering, because PostgreSQL can return records randomly. Here is an [example from stackoverflow](https://stackoverflow.com/questions/9060875/activerecordqueryingfirst-returns-third-member-of-the-collection).